### PR TITLE
Veteran Profile Support

### PR DESCRIPTION
**Pull request by rickr**
Adds veteran profile support

Refs #137

### CODE REVIEW

**Comment by davidmolina** [comment]
🙌🏼🙌🏼🙌🏼

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -7,4 +7,8 @@ def ssl_configured?
     !Rails.env.production?
     !Rails.env.test?
   end
+
+  def after_sign_in_path_for(_resource)
```
Comment (nanoxd) [inline app/controllers/application_controller.rb]: You take in a `_resource` but don't use the argument. Is this supposed to be that way?

File: app/controllers/veterans/confirmations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+class Veterans::ConfirmationsController < Devise::ConfirmationsController
+  # GET /resource/confirmation/new
```
Comment (nanoxd) [inline app/controllers/veterans/confirmations_controller.rb]: Are the code comments included here useful? Are they a partial thrown away implementation?

File: app/controllers/veterans/omniauth_callbacks_controller.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+class Veterans::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  # You should configure your model like this:
```
Comment (nanoxd) [inline app/controllers/veterans/omniauth_callbacks_controller.rb]: Are the code comments included here useful? Are they a partial thrown away implementation?

File: app/controllers/veterans/passwords_controller.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
+class Veterans::PasswordsController < Devise::PasswordsController
+  # GET /resource/password/new
```
Comment (nanoxd) [inline app/controllers/veterans/passwords_controller.rb]: > Are the code comments included here useful? Are they a partial thrown away implementation?

File: app/controllers/veterans/sessions_controller.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
+class Veterans::SessionsController < Devise::SessionsController
+# before_filter :configure_sign_in_params, only: [:create]
```
Comment (nanoxd) [inline app/controllers/veterans/sessions_controller.rb]: > Are the code comments included here useful? Are they a partial thrown away implementation?

File: app/controllers/veterans/unlocks_controller.rb [hunk 6]
```diff
@@ -0,0 +1,28 @@
+class Veterans::UnlocksController < Devise::UnlocksController
+  # GET /resource/unlock/new
```
Comment (nanoxd) [inline app/controllers/veterans/unlocks_controller.rb]: > Are the code comments included here useful? Are they a partial thrown away implementation?

File: app/models/veteran.rb [hunk 7]
```diff
@@ -13,10 +13,17 @@
 #
 
 class Veteran < ActiveRecord::Base
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable
   EMAIL_REGEX = /\A[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]+\z/
```
Comment (nanoxd) [inline app/models/veteran.rb]: I'd recommend reading up on [e-mail regexes](http://www.regular-expressions.info/email.html) for the lolz. There are a few variations of official RFC 5322 e-mail addresses that this regex does not support.

File: app/views/layouts/_header.html.erb [hunk 8]
```diff
@@ -1,95 +1,57 @@
-<!-- <nav class="navbar navbar-default navbar-fixed-top">
-  <div class="container-fluid">
+<!-- Fixed navbar -->
+<nav class="navbar navbar-default navbar-fixed-top">
+  <div class="container">
     <div class="navbar-header">
-      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
+      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false" aria-controls="navbar">
```
Comment (nanoxd) [inline app/views/layouts/_header.html.erb]: 👍 for accessibility

File: app/views/veterans/profile/_announcements.html.erb [hunk 9]
```diff
@@ -0,0 +1,3 @@
+<% if @announcements.nil? %>
+  <center><i>No announcements</i></center>
```
Comment (nanoxd) [inline app/views/veterans/profile/_announcements.html.erb]: Per [i](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/i#Notes)'s documentation, use newer tags or CSS to italicize the text.

File: app/views/veterans/profile/_mentor.html.erb [hunk 10]
```diff
@@ -0,0 +1,3 @@
+<% if @mentor.nil? %>
+  <center><i>You have not requested a mentor</i></center>
```
Comment (nanoxd) [inline app/views/veterans/profile/_mentor.html.erb]: Per [i's](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/i#Notes) documentation, use newer tags or CSS to italicize the text. 

[center](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/center) has been deprecated.

File: app/views/veterans/show.html.erb [hunk 11]
```diff
@@ -1,4 +1,10 @@
-<p id="notice"><%= notice %></p>
+<div class="container bs-docs-container">
+  <div class="row">
+    <div class="col-md-12" role="main">
+      <p id="notice"><%= notice %></p>
+
+      Hello my name is <%= @veteran.first_name %>
+      Hello my name is <%= @veteran.id %>
```
Comment (nanoxd) [inline app/views/veterans/show.html.erb]: The same text is repeated twice with different instance variables.

File: spec/models/veteran_spec.rb [hunk 12]
```diff
@@ -60,29 +57,30 @@
...
-          merge_fields: {
-            FNAME: veteran.first_name,
-            LNAME: veteran.last_name
-          }
-        }
-      )
-
-      veteran.add_to_mailchimp
-    end
+    # TODO: Move these to an active job test
```
Comment (nanoxd) [inline spec/models/veteran_spec.rb]: TODO spotted 👀

**Comment by nanoxd** [comment]
Made a few comments in a few areas. Please correct the html tags to ensure compatibility with modern browsers. Otherwise, good job 🙌

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (rickr) [inline app/controllers/application_controller.rb]: I believe so - this is a callback from devise.

File: app/views/veterans/show.html.erb (hunk 11 above)
Comment (rickr) [inline app/views/veterans/show.html.erb]: This is actually dead code. Good catch!

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (nanoxd) [inline app/controllers/application_controller.rb]: 👌

**Pull request merged**